pyzo/util/paths.py

```diff
@@ -130,18 +136,36 @@ def appdata_dir(appname=None, roaming=False, macAsLinux=False):
             else:
                 path = localpath
                 break
-    
+    data_path, config_path = path, path
+
     # Get path specific for this app
     if appname:
         if path == userDir:
             appname = '.' + appname.lstrip('.') # Make it a hidden directory
         path = os.path.join(path, appname)
+        data_path, config_path = path, path
+
         if not os.path.isdir(path):
-            os.mkdir(path)
-    
-    # Done
-    return path
+            # Better way to get config/data folder especially on *nix system (see XDG_CONFIG_HOME standard),
+            # but this should work on any os.
+            # For retro-compatibility, check if old folder exist, and if not, use standard path.
+            standard_data_path = QtCore.QStandardPaths.writableLocation(QtCore.QStandardPaths.AppDataLocation)
+            standard_config_path = QtCore.QStandardPaths.writableLocation(QtCore.QStandardPaths.ConfigLocation)
+
+            # Check if QStandardPaths succeeded to find the location, otherwise use old path
+            if standard_config_path != "" and standard_data_path != "":
```

Comment:
Maybe simply `if not (standard_config_path and standard_data_path):`, that also feels a bit safer  in case Qt ever returns `None`.